utils/signature/sigtest/endorser.go

```diff
@@ -0,0 +1,210 @@
+	messageEndorserUsingKeySigner struct {
+		signer interface {
+			Sign(signature.Digest) (signature.Signature, error)
+		}
+	}
+)
+
+var dummyEndorsement = CreateEndorsementsForThresholdRule(make([]byte, 0))[0]
+
+// NewNsEndorser creates a new namespace endorser according to the implementation scheme.
+func NewNsEndorser(scheme signature.Scheme, key []byte) (*NsEndorser, error) {
+	scheme = strings.ToUpper(scheme)
+	var err error
+	ret := &NsEndorser{}
+	switch scheme {
+	case signature.NoScheme, "":
+		ret.MessageEndorser = nil
+	case signature.Ecdsa:
+		signingKey, parseErr := ParseSigningKey(key)
+		err = parseErr
+		ret.MessageEndorser = &messageEndorserUsingKeySigner{signer: &ecdsaSigner{signingKey: signingKey}}
+	case signature.Bls:
+		sk := big.NewInt(0)
+		sk.SetBytes(key)
+		ret.MessageEndorser = &messageEndorserUsingKeySigner{signer: &blsSigner{sk}}
+	case signature.Eddsa:
+		ret.MessageEndorser = &messageEndorserUsingKeySigner{signer: &eddsaSigner{PrivateKey: key}}
+	default:
+		return nil, errors.Newf("scheme '%v' not supported", scheme)
+	}
+	return ret, err
+}
+
+// NewMspEndorser creates a new MspEndorser given MSP directories.
+// This signer will create an endorsement for each MSP provided.
+func NewMspEndorser(certType int, mspDirs ...msp.DirLoadParameters) (*NsEndorser, error) {
+	identities, err := GetSigningIdentities(mspDirs...)
+	if err != nil {
+		return nil, err
+	}
+	mspEndorser := &MspEndorser{
+		certType:   certType,
+		identities: identities,
+		mspIDs:     make([][]byte, len(mspDirs)),
+		certsBytes: make([][]byte, len(mspDirs)),
+	}
+	for i, id := range identities {
+		mspEndorser.mspIDs[i] = []byte(id.GetMSPIdentifier())
+		serializedIDBytes, err := id.Serialize()
+		if err != nil {
+			return nil, errors.Wrap(err, "serializing default signing identity")
+		}
+		serializedID, err := protoutil.UnmarshalSerializedIdentity(serializedIDBytes)
+		if err != nil {
+			return nil, err
+		}
+		idBytes := serializedID.IdBytes
+		if certType == test.CreatorID {
+			idBytes, err = DigestPemContent(idBytes, bccsp.SHA256)
+			if err != nil {
+				return nil, err
+			}
+		}
+		mspEndorser.certsBytes[i] = idBytes
+	}
+	return &NsEndorser{MessageEndorser: mspEndorser}, nil
+}
+
+// GetSigningIdentities loads signing identities from the given MSP directories.
+func GetSigningIdentities(mspDirs ...msp.DirLoadParameters) ([]msp.SigningIdentity, error) {
+	identities := make([]msp.SigningIdentity, len(mspDirs))
+	for i, mspDir := range mspDirs {
+		localMsp, err := msp.LoadLocalMspDir(mspDir)
+		if err != nil {
+			return nil, err
+		}
+		identities[i], err = localMsp.GetDefaultSigningIdentity()
+		if err != nil {
+			return nil, errors.Wrap(err, "loading signing identity")
+		}
+	}
+	return identities, nil
+}
+
+// EndorseTxNs endorses a transaction's namespace.
+func (v *NsEndorser) EndorseTxNs(txID string, tx *applicationpb.Tx, nsIndex int) (*applicationpb.Endorsements, error) {
```

Comment:
Please group methods for the same struct together. Mixing them reduces readability.